Update Transaction.php to fix transaction amount error

git-seb opened the pull request:
Fix: wrong amount was being sent to API causing the following error showing up: 'Something went wrong: Transaction amount must be greater than 0'

Crypto2 commented:
Without conversion this looks like it could result in a mix of currencies. ie. if the user picked Japanese Yen but the store's internal currency is USD.

git-seb commented:
> Without conversion this looks like it could result in a mix of currencies. ie. if the user picked Japanese Yen but the store's internal currency is USD.

Without this change, it sent the wrong amount.

See here log before change:
    [amount] => 0.00101081 
    [currency1] => GBP 
    [currency2] => BTC 
    [buyer_email] => [xxxx](mailto:xxx@xxx.com) 
    [buyer_name] => Seb 
    [invoice] => ABSDEV_202400114 
    [ipn_url] => https://dev.xxx.co.uk/en/coinpayments/ipn/handle/
    
Log after change:
    [amount] => 8.0000
    [currency1] => GBP
    [currency2] => BTC
    [buyer_email] => [xxxx](mailto:xxx@xxx.com) 
    [buyer_name] => Seb 
    [invoice] => ABSDEV_202400123
    [ipn_url] => https://dev.xxx.co.uk/en/coinpayments/ipn/handle/

Crypto2 commented:
I mainly just mean that it needs to be sure that these are both referring to the same currency, no matter what currency the user  has selected:

```
'amount' => $order->getBaseGrandTotal(),
'currency1' => $this->getPaymentConfig('receive_currency'),
```

So they can't just pick JPY and get a massive discount or anything like that. So I think if you are using getBaseGrandTotal() then currency1 should probably be getBaseCurrencyCode() if these are the right docs https://www.magentoextensions.org/documentation/interface_magento_1_1_quote_1_1_api_1_1_data_1_1_totals_interface.html

git-seb commented:
> I mainly just mean that it needs to be sure that these are both referring to the same currency, no matter what currency the user has selected:
> 
> ```
> 'amount' => $order->getBaseGrandTotal(),
> 'currency1' => $this->getPaymentConfig('receive_currency'),
> ```
> 
> So they can't just pick JPY and get a massive discount or anything like that. So I think if you are using getBaseGrandTotal() then currency1 should probably be getBaseCurrencyCode() if these are the right docs https://www.magentoextensions.org/documentation/interface_magento_1_1_quote_1_1_api_1_1_data_1_1_totals_interface.html

I tested your suggestion but it thrown an error (Transaction is not found). Actually when that error happened, the error function actually contains an error as well (error is undefined) I've fixed that as well which I will add later as well.

So I reverted to before your suggestion to the fix I provided in this pull request, changed the webshop to EUR and tested placing an order. The EUR amount in cart was €9,39. After placing the order I see in CoinPaymentDebug.log it still picked GBP instead of EUR so that is perfect. It also took the GBP amount of 8,00 so it did not resulted in mixed currencies.
